docs/functionality-overview.md

```diff
@@ -0,0 +1,184 @@
+# Functionality overview
+
+The paste HTML to govspeak functionality interfaces with data on a users
+clipboard to attempt to create govspeak formatted text at the point of them
+pasting. The expectation for usage is that users may be copying data from a
+document created on a word processor (such as Microsoft Word or Google Docs),
+fixed layout documents (such as PDF) or pasting content from webpages.
+
+The quality and accuracy of clipboard data can vary significantly which means
+that the outcome can sometimes be surprising. The goal of the tool is
+therefore somewhat modest: to consistently provide a paste experience that is
+at least as good as pasting plain text and to enhance wherever possible.
+
+This document serves to explain more about how this tool works (or
+doesn't) and at look some of the issues discovered in testing scenarios.
+
+## How this tool works
+
+This tools listens for a paste event to occur on a designated element. When
+this event occurs it uses the [ClipboardData][] API to check whether the paste
+content is available in a HTML format and, if so, converts this HTML to
+Govspeak. If HTML is not available it falls back to the browsers default
+behaviour - which would be to insert or replace the text content of the paste.
+
+As HTML is a widely understood open format it is frequently available as part
+of clipboard data to allow the communication of rich text between applications.
+For HTML to be available on a paste event it needs to have been written to the
+clipboard at the time it was copied. This is something that is done by the
+application displaying the content and may use features of the underlying
+operating system. For example, if you copy text in Microsoft Word it will
+automatically convert this text to HTML (and other formats such as plain text)
+when placing it on the clipboard.
+
+The conversion of HTML to Govspeak is relatively simple. Elements that can be
+identified as having a Govspeak equivialent, such as a link, are converted to
+their equivalent; elements that are used for non visible data, such as style
+information, are removed; and any other elements are converted to their text
+representation. This conversion is done by using the [turndown] package.
+
+In order to produce a good Govspeak conversion the HTML must be marked up
+[semantically][semantic-html]. This allows the original authors content intent
+to be understood by different application regardless of the formatting. In
+order for this to occur the author must have specified these semantics when
+producing their document and the application used to read the document must
+include this information in copy data to the clipboard.
+
```

Comment:
It might be helpful to include an example here e.g. the user formats text as a heading which enables the application to convert it to a heading in HTML/

Reply:
After talking to Sonia (asking about periods or no periods in e.g.) I've gone with "for example"